examples/vision/attention_mil_classification.py

```diff
@@ -0,0 +1,589 @@
+
+    return keras.Model(inputs, output)
+
+
+"""
+## Class weights
+
+Since this kind of problem could simply turn into imbalanced data classification problem,
+class weights should be considered.
+
+Let's say there are 1000 bags. There often could be cases were ~90 % of the bags does not
+contain any positive label and ~10 % does.
+Such data can be referred to as **Imbalanced data**.
+
+Using class weights, the model will tend to consider the rare class more as compared to
+that of the abundant one.
+"""
+
+
+def compute_class_weights(labels):
+
+    # Count number of postive and negative bags.
+    negative_count = len(np.where(labels == 0)[0])
+    positive_count = len(np.where(labels == 1)[0])
+    total_count = negative_count + positive_count
+
+    # Build class weight dictionary.
+    return {
+        0: (1 / negative_count) * (total_count / 2),
+        1: (1 / positive_count) * (total_count / 2),
+    }
+
+
+"""
+## Build and train model
+
+The model is built and trained in this section.
+
+Some regularization techniques are considered to avoid overfitting the model which ensures
+minimal generalization error.
+"""
+
+
+def train(train_data, train_labels, val_data, val_labels, model):
+
+    # Train model.
+    # Prepare callbacks.
+    # Path where to save best weights.
+
+    # Take the file name from the wrapper.
+    file_path = "/tmp/best_model_weights.h5"
+
+    # Initialize model checkpoint callback.
+    model_checkpoint = keras.callbacks.ModelCheckpoint(
+        file_path,
+        monitor="val_loss",
+        verbose=0,
+        mode="min",
+        save_best_only=True,
+        save_weights_only=True,
+    )
+
+    # Initialze early stopping callback.
+    # The model performance is monitored across the unseen data and stops training
+    # when the generalization error cease to decrease.
+    early_stopping = keras.callbacks.EarlyStopping(
+        monitor="val_loss", patience=10, mode="min"
+    )
+
+    # Compile model.
+    model.compile(
+        optimizer="adam", loss="sparse_categorical_crossentropy", metrics=["accuracy"],
+    )
+
+    # Fit model.
+    model.fit(
+        train_data,
+        train_labels,
+        validation_data=(val_data, val_labels),
+        epochs=20,
+        class_weight=compute_class_weights(train_labels),
+        batch_size=1,
+        callbacks=[early_stopping, model_checkpoint],
+        verbose=0,
+    )
+
+    # Load best weights.
+    model.load_weights(file_path)
+
+    return model
+
+
+# Building model(s).
+instance_shape = train_data[0][0].shape
+models = [create_model(instance_shape) for _ in range(ENSEMBLE_AVG_COUNT)]
+
+# Show single model architecture.
+print(models[0].summary())
+
+# Training model(s).
+trained_models = [
+    train(train_data, train_labels, val_data, val_labels, model)
+    for model in tqdm(models)
+]
+
+"""
+## Model evaluation
+
+The models are in their deterministic state now and ready for evaluation.
```

Comment:
deterministic state? why not just say

The models are now and ready for evaluation.

Reply:
Updated it.